User searches for drinks

clalil opened the pull request:
# Search for drink functionality
[Link to PT](https://www.pivotaltracker.com/story/show/168979772)

# PR content
In this PR we are adding a search functionality using a mock api response.

# This PR contains
Feature tests
Basic styling
A search functionality
A mock API response

# What I have learned
How to mock an API response in Rails
How to add a simple search functionality
How to do basic styling using tailwind

clalil committed:
Add drink list feature and required step definitions.

lucamarial committed:
Merge branch 'display_cocktail' of https://github.com/clalil/prestige_worldwide into display_cocktail

lucamarial committed:
Generates Cocktails controller and adds root_path to step def

miyeaier committed:
addit Servicefolder and file ,install gem rest-client

clalil committed:
Add JSON parsed variable returning drinks  for API future mock call.

clalil committed:
Rephrase steps for search feature.

lucamarial committed:
Refactores cocktails_controller

Adds get drink and get image method to service file.
Adds cocktail and and cocktail_image variable to html.

clalil committed:
Install webmock, modify controller and mock api response
Configure feature test and steps, configure service model for user input in search field
Add search field and ternary code for nil user input on initial page loading
Add mock file for API.

lucamarial committed:
Adds error message to html and elsif statement to index and controller. Adds sad path.

clalil committed:
Fix bug from line 17 in CocktailService module, adjusting ternary operator for empty string during sad path.

faraznaeem commented:
Is this feature still WIP

clalil committed:
Troubleshooting code.

lucamarial committed:
Adds scenario to search for multible ingredients

clalil committed:
Solve merge.

clalil committed:
Corrects bugged code, revert to previous.

lucamarial committed:
Adds div to html and css styling

lucamarial committed:
Solves merge conflict and adds styling

clalil committed:
Solves merge conflict.

clalil committed:
Update steps and feature to fix test-related routing bug due to merge conflict previously.
Correct space in index file.

clalil committed:
Correct typo in scenario.

oliverochman commented on config/routes.rb:
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   root controller: :welcome, action: :index
```
why do you have this route?

clalil replied:
Originally, me and Luca created a root controller for the cocktail page. This is the route that was merged from the guy's previous PR. Since our root controller's overlapped we kept theirs since ours could just be used as a resource.

oliverochman replied:
The main functionality of the app is this feature.
Does the other welcome controller have any functionality?

clalil replied:
Not that I am aware of. But I thought we always had to have a root controller for at least the main page?

oliverochman commented on features/search_for_cocktails.feature:
```diff
@@ -0,0 +1,17 @@
+  I need to see the name of the returned drink
+
+  Scenario: Successfully searches for one ingredient
+    Given I am on the cocktail page
```
Can the user get to this page if he visits the root_path first?

clalil replied:
Not unless he knows which route to manually pass into the url. This page will be available later only once you've signed in and paid.

oliverochman replied:
Premature optimization, please make the cocktail index action the root_path

clalil replied:
But our cocktail page is on the cocktails_path and not on the welcome_path...

oliverochman commented on features/search_for_cocktails.feature:
```diff
@@ -0,0 +1,17 @@
+    Then I should see '3-Mile Long Island Iced Tea'
+
+  Scenario: User can't search
+    Given I am on the cocktail page
```
Please add a background where you say where the user is instead of repeating this step in two scenarios

clalil replied:
Changed accordingly.

oliverochman commented on features/search_for_cocktails.feature:
```diff
@@ -0,0 +1,17 @@
+
+  Scenario: Successfully searches for one ingredient
+    Given I am on the cocktail page
+    And I fill in 'Search' with 'Gin'
```
`When I fill in 'Search' with 'Gin'`

clalil replied:
Changed accordingly.

clalil committed:
Add background to feature and change step defintion.

clalil committed:
Delete unneccessary routes and controllers
Correct step definitions and add background. Merge welcome and cocktails page into one main page.

clalil committed:
Rename welcome page feature.

oliverochman commented on app/views/cocktails/index.html.haml:
```diff
@@ -1,3 +1,15 @@
+%h1 Search for your drink of choice: 
```
shouldn't this be underneath the title at line 13

clalil replied:
Changed accordingly.

clalil committed:
Remove search functionality to below main page title.
